### PR TITLE
Added base Conda environment yml file

### PR DESCRIPTION
**Pull request by jdye64**

### CODE REVIEW

**Comment by jdye64** [comment]
I think more docs is always great! Specifically which doc building things are you referring? I can add them pretty quickly if you can point me to them.

**Comment by CJ-Wright** [comment]
https://github.com/python-streamz/streamz/blob/master/docs/requirements-docs.txt

**Comment by jdye64** [comment]
Ah, yes I misunderstood what you were talking about. I will add those now.

File: conda/environments/streamz_dev.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: streamz_dev
+channels:
+  - conda-forge
+  - defaults
+dependencies:
+  - python>=3.7
+  - cython>=0.29,<0.30
```
Comment (CJ-Wright) [inline conda/environments/streamz_dev.yml]: Where is the cython dep coming from?

**Comment by CJ-Wright** [comment]
@martindurant thoughts?

File: conda/environments/streamz_dev.yml (hunk 1 above)
Comment (martindurant) [reply conda/environments/streamz_dev.yml]: cytoolz? I don't suppose we're building anything.

**Comment by martindurant** [comment]
I'm generally OK with the idea, but wonder whether the CI tests should explicitly use the same environment.

File: conda/environments/streamz_dev.yml (hunk 1 above)
Comment (CJ-Wright) [reply conda/environments/streamz_dev.yml]: Right, if one of the deps needs it conda should figure that out.

Comment (jdye64) [reply conda/environments/streamz_dev.yml]: ahh ... my bad I meant to remove that. I was experimenting with implementing librdkafka with cython and forgot to remove it. I will remove them now.

On a side note that is working very well BTW and might open something up around that as well. Working on the best way to make cython not a hard dependency at the moment since I'm sure several people don't using streamz don't necessarily want it.

Comment (martindurant) [reply conda/environments/streamz_dev.yml]: > Working on the best way to make cython not a hard dependency

Typically you can include the C file for those that want to compile (source dist) and of course, if you build wheels and conda packages, those should not need cython.

Comment (CJ-Wright) [reply conda/environments/streamz_dev.yml]: I'm confused, do you mean that you are using librdkafka with cython or using cython to speed up `streamz` kafka bindings?

Comment (jdye64) [reply conda/environments/streamz_dev.yml]: > I'm confused, do you mean that you are using librdkafka with cython or using cython to speed up `streamz` kafka bindings?

Actually both. I was replacing the current kafka sources with my own sources (not the confluent-python-kafka library) which were cython modules that interacted directly with librdkafka++. There are many many advantages to that especially in the realm of partition balancing that the new KafkaConsumer offers (not implemented in the confluent python library yet) for distributed systems with many processes appearing/disappearing.

In a nutshell I was seeing about 15-20X speedups in my consumers when using my cython implementations.

**Comment by CJ-Wright** [comment]
@jdye64 would you be interested in changing the `.travis.yaml` to use the `environment.yml` file to build the testing env?

**Comment by jdye64** [comment]
Sure. I noticed there are a few other dependencies in the .travis.yml file that were being installed so I'm going to add those to the environment.yml file as well.

**Comment by jdye64** [comment]
@CJ-Wright Ok had to step away for a bit but looks like its all good to go and using the conda environment file correctly in Travis now.

**Comment by CJ-Wright** [comment]
Thank you very much!
